aspnet/fundamentals/hosting.rst

```diff
@@ -0,0 +1,226 @@
+:version: 1.0.0
+
+Hosting
+=======
+
+By `Steve Smith`_
+
+To run an ASP.NET Core app, you need to configure and launch a host using `WebHostBuilder <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Hosting/WebHostBuilder/index.html?highlight=webhostbuilder>`_.
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+What is a Host?
+---------------
+
+ASP.NET Core apps require a *host* in which to execute. A host must implement the :dn:iface:`~Microsoft.AspNetCore.Hosting.IWebHost` interface, which exposes collections of features and services, and a ``Start`` method. The host is typically created using an instance of a :dn:class:`~Microsoft.AspNetCore.Hosting.WebHostBuilder`, which builds and returns a  :dn:class:`~Microsoft.AspNetCore.Hosting.Internal.WebHost` instance. The ``WebHost`` references the server that will handle requests. Learn more about :doc:`servers <servers>`.
+
+What is the difference between a host and a server?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The host is responsible for application startup and lifetime management. The server is responsible for accepting HTTP requests. Part of the host's responsibility includes ensuring the application's services and the server are available and properly configured. You can think of the host as being a wrapper around the server. The host is configured to use a particular server; the server is unaware of its host.
+
+Setting up a Host
+-----------------
+
+You create a host using an instance of ``WebHostBuilder``. This is typically done in your app's entry point, which by default will be located in a *Program.cs* file. A typical *Program.cs*, shown below, demonstrates how to use a ``WebHostBuilder`` to build a host. 
```

Comment:
Maybe it's more accurate to say that the entry point is located in the `public static void Main()` function? Since it doesn't really matter which class or file the entry point is located in. Would be fine to leave it as is as well though.